src/lib/DisposableManager.ts

```diff
@@ -1,76 +1,24 @@
-import DisposableTypes from './data/DisposableTypes';
-import { IDisposable } from './interface/IDisposable';
-
 /**
- * Utility class for handling various disposable things like timeouts, disposable instances.
- *
- * DisposableManager.register('interval', (interval) => {
- *  clearInterval(interval);
- * );
+ * Utility class for handling various disposable things like timeouts, disposable functions.
  *
  * @class DisposableManager
  * @constructor
  */
 export default class DisposableManager {
-  /**
-   * @type {{}}
-   */
-  public static types: { [key: string]: Function } = {
-    [DisposableTypes.DISPOSABLE]: (disposable: IDisposable) => {
-      disposable.dispose();
-    },
-    [DisposableTypes.INTERVAL]: (interval: number) => {
-      clearInterval(interval);
-    },
-  };
   /**
    * list of disposables, like Event handlers, used for destruction
    * @type {{}}
    */
-  private disposables: { [key: string]: Array<IDisposable | any> } = {};
-
-  /**
-   * @method register
-   * @param {string} type
-   * @param {Function} dispose
-   */
-  public static register(type: string, dispose: Function): void {
-    DisposableManager.types[type] = dispose;
-  }
+  private readonly disposables: Array<(() => void)> = [];
 
   /**
    *
    * @method add
    * @param disposable
-   * @returns {any}
-   */
-  public add(disposable: any, type: string = DisposableTypes.DISPOSABLE): any {
-    if (!DisposableManager.types[type]) {
-      throw new Error(`Following type is not registered yet: [${type}]`);
-    }
-
-    if (!this.disposables) {
-      this.disposables = {};
-    }
-
-    if (!this.disposables[type]) {
-      this.disposables[type] = [];
-    }
-
-    this.disposables[type].push(disposable);
-
-    return disposable;
-  }
-
-  /**
-   * Adds an interval
-   *
-   * @method addInterval
-   * @param interval {number} id of the interval
-   * @returns {number} id of the interval
+   * @returns void
    */
-  public addInterval(interval: number): number {
-    return this.add(interval, DisposableTypes.INTERVAL);
+  public add(disposable: () => void): void {
```

Comment:
Should we create an `Disposer` type for `() => void`? (and change `disposable` to `disposer` either way)

So it can be reused:
* In the `disposeAll` helper function (from the other comment)
* in user code, or their own custom helpers (functions that should return an `Disposer` so they can be passed here.

Doesn't change the code, but could convey purpose.

Reply:
If the interface was a bit more complex than just a function without arguments I would say yes, but in this case I do not see a benefit on doing that.